OpenRA.Game/HotkeyManager.cs

```diff
@@ -0,0 +1,103 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using FS = OpenRA.FileSystem.FileSystem;
+
+namespace OpenRA
+{
+	public sealed class HotkeyDefinition
+	{
+		public readonly string Name;
+		public readonly Hotkey Default = Hotkey.Invalid;
+		public readonly string Description = "";
+		public readonly HashSet<string> Types = new HashSet<string>();
+
+		public HotkeyDefinition(string name, MiniYaml node)
+		{
+			Name = name;
+
+			if (!string.IsNullOrEmpty(node.Value))
+				Default = FieldLoader.GetValue<Hotkey>("value", node.Value);
+
+			var descriptionNode = node.Nodes.FirstOrDefault(n => n.Key == "Description");
+			if (descriptionNode != null)
+				Description = descriptionNode.Value.Value;
+
+			var typesNode = node.Nodes.FirstOrDefault(n => n.Key == "Types");
+			if (typesNode != null)
+				Types = FieldLoader.GetValue<HashSet<string>>("Types", typesNode.Value.Value);
+		}
+	}
+
+	public sealed class HotkeyManager
+	{
+		readonly Dictionary<string, Hotkey> settings;
+		readonly Dictionary<string, HotkeyDefinition> definitions = new Dictionary<string, HotkeyDefinition>();
+		readonly Dictionary<string, Hotkey> keys = new Dictionary<string, Hotkey>();
+
+		public HotkeyManager(FS fileSystem, Dictionary<string, Hotkey> settings, Manifest manifest)
```

Comment:
This can be `IReadOnlyFileSystem fileSystem`, and then `using FS = OpenRA.FileSystem.FileSystem;` can go away.